pkg/monitoring/metrics/forklift-controller/migration_metrics.go

```diff
@@ -68,52 +66,54 @@ func RecordMigrationMetrics(c client.Client) {
 
 				provider := sourceProvider.Type().String()
 
-				if m.Status.HasCondition(Succeeded) {
-					key = fmt.Sprintf("%s|%s|%s|%s|%s", Succeeded, provider, mode, target, string(plan.UID))
-					counterMap[key]++
+				statusList := []string{Succeeded, Failed, Executing, Canceled}
+				for _, status := range statusList {
+					if m.Status.HasCondition(status) {
```

Comment:
we can't have more than a single condition that correlates with the above statuses, right?  in that case, we can optimize the code a bit

Reply:
I've did some refactoring and tried to simply this as much as possible.